src/mem/mem_ctrl.cc

```diff
@@ -302,10 +460,90 @@ MemCtrl::addToReadQueue(PacketPtr pkt, unsigned int pkt_count, bool is_dram)
     // queue, do so now
     if (!nextReqEvent.scheduled()) {
         DPRINTF(MemCtrl, "Request scheduled immediately\n");
+        //COMMENT: DRAM cache tag check latency should be added to the
+        // curTick()
+        schedule(nextReqEvent, curTick());
+        //COMMENT: tags will be checked on the way back
+    }
+}
+
+void
+MemCtrl::addToDRAMFillQueue(MemPacket mem_pkt)
```

Comment:
```suggestion
MemCtrl::addToDRAMFillQueue(const MemPacket *mem_pkt)
```

Check on pointer vs not pointer.